codecparsers/bitReader.h

```diff
@@ -85,7 +85,7 @@ bool BitReader::readT(T& v, uint32_t nbits)
     uint32_t tmp;
     if (!read(tmp, nbits))
         return false;
-    v = tmp;
+    v = static_cast <T>(tmp);
```

Comment:
Little wired, `bool BitReader::readT(bool& v)` will pass bool to `bool BitReader::readT(T& v, uint32_t nbits)` too.
I did not see error in my machine. 
From this http://www.cplusplus.com/doc/tutorial/typecasting/ uint32_t implicate to bool is no problem.